pootle/apps/pootle_config/getters.py

```diff
@@ -0,0 +1,44 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) Pootle contributors.
+#
+# This file is a part of the Pootle project. It is distributed under the GPL3
+# or later license. See the LICENSE file for a copy of the license and the
+# AUTHORS file for copyright and authorship information.
+
+from django.utils.translation import ugettext_lazy as _
+
+from pootle.core.delegate import config
+from pootle.core.plugin import getter
+
+from .exceptions import ConfigurationError
+from .models import Config
+
+
+@getter(config)
+def config_getter(sender=None, instance=None, key=None, **kwargs):
+
+    if sender:
+        if instance is not None and not isinstance(instance, sender):
+            raise ConfigurationError(
+                _("'instance' must be an instance of sender, when specified"))
+        if instance:
+            conf = Config.objects.config_for(instance)
+        else:
+            conf = Config.objects.config_for(sender).filter(object_pk__isnull=True)
+    elif instance:
+        raise ConfigurationError(
+            _("'instance' must be an instance of sender, when specified"))
```

Comment:
wierd, i was just looking at the same bit of code. I think this could be refactored a bit 8/